Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/skylanders/ui/SkylanderSlotAdapter.java

```diff
@@ -0,0 +1,159 @@
+package org.dolphinemu.dolphinemu.features.skylanders.ui;
+
+import android.content.Intent;
+import android.util.Pair;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AlertDialog;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.google.android.material.dialog.MaterialAlertDialogBuilder;
+
+import org.dolphinemu.dolphinemu.R;
+import org.dolphinemu.dolphinemu.activities.EmulationActivity;
+import org.dolphinemu.dolphinemu.databinding.DialogCreateSkylanderBinding;
+import org.dolphinemu.dolphinemu.databinding.ListItemSkylanderSlotBinding;
+import org.dolphinemu.dolphinemu.features.skylanders.SkylanderConfig;
+import org.dolphinemu.dolphinemu.features.skylanders.model.Skylander;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SkylanderSlotAdapter extends RecyclerView.Adapter<SkylanderSlotAdapter.ViewHolder>
+        implements AdapterView.OnItemClickListener
+{
+  public static class ViewHolder extends RecyclerView.ViewHolder
+  {
+    public ListItemSkylanderSlotBinding binding;
+
+    public ViewHolder(@NonNull ListItemSkylanderSlotBinding binding)
+    {
+      super(binding.getRoot());
+      this.binding = binding;
+    }
+  }
+
+  private final List<SkylanderSlot> mSlots;
+  private final EmulationActivity mContext;

```

Comment:
Since it's typed as `EmulationActivity` and you're calling `EmulationActivity` methods on it, this would be better named `mActivity`.